### PR TITLE
Refactor DirectEmbedding

### PR DESCRIPTION
**Pull request by olafurpg**
* Move LiftingTransformer into separate ReifyAsEmbedding
  module, allowing more flexibility for adding other kinds
  of transformations
* Improve error handling for missing reifyAs annotations
* Add support for blocks
* Improve minor wording issues in README

### CODE REVIEW

**Comment by vjovanov** [comment]
Review by @vjovanov

File: dsls/src/test/scala/ch/epfl/directembedding/test/BasicSpec.scala [hunk 1]
```diff
@@ -158,4 +161,19 @@ class BasicSpec extends FlatSpec with ShouldMatchers {
...
+
+  "lift" should "take the last expression in blocks" in {
+    testReify(implicit collec =>
+      lift {
+        println("debug")
+        ObjectExample.valDef
+      }) should be(List(ValDef))
+  }
+
+  //  "lift" should "give a friendly error when a reifyAs annotation is missing" in {
```
Comment (vjovanov) [inline dsls/src/test/scala/ch/epfl/directembedding/test/BasicSpec.scala]: Maybe we can open a ticket for this.

File: dsls/src/test/scala/ch/epfl/directembedding/test/BasicSpec.scala [hunk 2]
```diff
@@ -127,6 +129,7 @@ class BasicSpec extends FlatSpec with ShouldMatchers {
   "lift" should "work with TArgClassExample methods with app 1 args" in {
     testReify(implicit collec =>
       lift {
+
```
Comment (vjovanov) [inline dsls/src/test/scala/ch/epfl/directembedding/test/BasicSpec.scala]: Bad formatting.

File: dsls/src/test/scala/ch/epfl/directembedding/test/BasicSpec.scala [hunk 3]
```diff
@@ -90,6 +91,7 @@ class BasicSpec extends FlatSpec with ShouldMatchers {
   }
 
   "lift" should "work with TArgClassExample methods with take" in {
+
```
Comment (vjovanov) [inline dsls/src/test/scala/ch/epfl/directembedding/test/BasicSpec.scala]: Bad formatting.

File: dsls/src/test/scala/ch/epfl/directembedding/test/BasicSpec.scala [hunk 4]
```diff
@@ -52,6 +52,7 @@ class BasicSpec extends FlatSpec with ShouldMatchers {
       lift {
         ObjectExample.nested.valDef
       }) should be(List(ValDef))
+
```
Comment (vjovanov) [inline dsls/src/test/scala/ch/epfl/directembedding/test/BasicSpec.scala]: Bad formatting.

**Comment by vjovanov** [comment]
Otherwise, LGTM.

**Comment by olafurpg** [comment]
Formatting fixed @vjovanov

**Pull request merged**